controllers/imageclusterinstall_controller.go

```diff
@@ -753,35 +773,57 @@ func (r *ImageClusterInstallReconciler) writeInputData(
 			return err
 		}
 		clusterInfo := r.getClusterInfoFromFile(clusterInfoFilePath)
+		if clusterInfo == nil {
+			clusterInfo = &lca_api.SeedReconfiguration{}
+		}
 
 		crypto, err := r.Credentials.EnsureKubeconfigSecret(ctx, cd, clusterInfo)
 		if err != nil {
 			return fmt.Errorf("failed to ensure kubeconifg secret: %w", err)
 		}
 
-		kubeadminPasswordHash, err := r.Credentials.EnsureAdminPasswordSecret(ctx, cd)
+		kubeadminPasswordHash, err := r.Credentials.EnsureAdminPasswordSecret(ctx, cd, clusterInfo.KubeadminPasswordHash)
 		if err != nil {
 			return fmt.Errorf("failed to ensure admin password secret: %w", err)
 		}
-		if err := r.writeClusterInfo(ctx, log, ici, cd, crypto, psData, kubeadminPasswordHash, clusterInfoFilePath, clusterInfo, bmh); err != nil {
-			return fmt.Errorf("failed to write cluster info: %w", err)
+
+		newClusterInfo, err := r.createSeedReconfiguration(ctx, log, ici, cd, crypto, psData, kubeadminPasswordHash, clusterInfo, bmh)
+		if err != nil {
+			return fmt.Errorf("failed to create seed reconfiguration: %w", err)
 		}
+		if !reflect.DeepEqual(newClusterInfo, clusterInfo) {
```

Comment:
Do we really need this check if we're checking the directory hash later?

Reply:
I believe we can remove it though what's the point to write file if we don't need but we do it for many other cases, so ill remove it